chimney/src/main/scala/io/scalaland/chimney/internal/compiletime/Contexts.scala

```diff
@@ -42,11 +44,22 @@ private[compiletime] trait Contexts { this: Definitions & Configurations =>
     }
     object ForTotal {
 
-      def create[From: Type, To: Type](src: Expr[From], config: TransformerConfig): ForTotal[From, To] =
+      def create[From: Type, To: Type](
+          src: Expr[From],
+          config: TransformerConfig,
+          runtimeDataStore: Option[Expr[TransformerDefinitionCommons.RuntimeDataStore]]
+      ): ForTotal[From, To] =
         ForTotal(
           From = Type[From],
           To = Type[To],
           src = src,
+          runtimeDataStore = runtimeDataStore
+            .getOrElse(
```

Comment:
There's an inconsistency between definition of context (where `runtimeDataStore` expr is required) and factory method (when it's actually optional). I'd strive for unifying so that we don't have to construct dummy expressions that accessing might be unsafe later.

Reply:
I was thinking about using temporary hack with something like:
```
  case q"""${something}.runtimeData""" => 
```
in legacy macros rule, to avoid big changes.

Reply:
See https://github.com/scalalandio/chimney/pull/299#discussion_r1183601495